src/model/index.ts

```diff
@@ -112,7 +118,12 @@ export function locationToView(location: HistoryLocation): CurrentView {
         view: 'search',
         term: fromNullable(location.search.term),
         location: fromNullable(location.search.location),
-        radius: IntFromString.decode(location.search.radius).fold(
+        radius: RadiusValue.decode(
+          IntFromString.decode(location.search.radius).fold(
+            _ => 0,
+            value => value
+          )
+        ).fold(
```

Comment:
there's a cleaner way?

Reply:
yes, since you are operating with the same monad (Either) for both `decode` functions, you can use `chain` (aka `flatMap` in Scala):
```suggestion
        radius: IntFromString.decode(location.search.radius)
          .chain(RadiusValue.decode)
          .fold(() => none, some)
```